File: Contents/임시파일_이건 지우자/MGESwitch/Sample/IosObjcSample/Supporting Files/Model/ItemsForTableView.m
//
//  ItemForTableView.m
//  CustomizingNavigationBar
//
//  Created by Kwan Hyun Son on 13/12/2018.
//  Copyright © 2018 Mulgrim Inc. All rights reserved.
//

#import "ItemsForTableView.h"
#import "Item.h"

@implementation ItemsForTableView

+ (instancetype)sharedItems {
    static ItemsForTableView *sharedItems = nil;
    static dispatch_once_t onceToken;          // dispatch_once_t는 long형
    dispatch_once(&onceToken, ^{
        sharedItems =  [[self alloc] initPrivate];
    });
    
    return sharedItems;
}



/// 진짜 초기화 메서드, 최초 1회만 호출될 것이다.
- (instancetype)initPrivate {
    self = [super init];
    if(self) {
        Item *item1 = [[Item alloc] initWithText:@"MGUDNSwitch" detailText:@"Day Night Switch"];

        Item *item2 = [[Item alloc] initWithText:@"MGUFlatSwitch" detailText:@"Flat Switch - V 체크"];

        Item *item3 = [[Item alloc] initWithText:@"MGUMaterialSwitch" detailText:@"MGUMaterialSwitch Basic"];
        Item *item4 = [[Item alloc] initWithText:@"MGUMaterialSwitch" detailText:@"MGUMaterialSwitch various examples"];

        Item *item5 = [[Item alloc] initWithText:@"MGUSevenSwitch" detailText:@"UISwitch와 같은 모양 연출가능하다."];
        Item *item6 = [[Item alloc] initWithText:@"MGUSevenSwitch" detailText:@"MGUSevenSwitch various examples"];

        Item *item7 = [[Item alloc] initWithText:@"MGUFavoriteSwitch" detailText:@"Favorite Switch Basic"];

        Item *item8 = [[Item alloc] initWithText:@"MGUOnOffButton" detailText:@"MGUOnOffButton Basic"];
        
        
        NSMutableDictionary <NSString *, NSArray <Item *>*>* section0 = @{
            @"MGUDNSwitch 섹션" :
        @[item1]
        }.mutableCopy;

        NSMutableDictionary <NSString *, NSArray <Item *>*>* section1 = @{
            @"MGUFlatSwitch 섹션" :
        @[item2]
        }.mutableCopy;

        NSMutableDictionary <NSString *, NSArray <Item *>*>* section2 = @{
            @"MGUMaterialSwitch 섹션" : @[item3, item4]
        }.mutableCopy;

        NSMutableDictionary <NSString *, NSArray <Item *>*>* section3 = @{
            @"MGUSevenSwitch 섹션" : @[item5, item6]
        }.mutableCopy;

        NSMutableDictionary <NSString *, NSArray <Item *>*>* section4 = @{
            @"MGUFavoriteSwitch 섹션" :
        @[item7]
        }.mutableCopy;

        NSMutableDictionary <NSString *, NSArray <Item *>*>* section5 = @{
            @"MGUOnOffButton 섹션" :
        @[item8]
        }.mutableCopy;

        _allItems = @[section0, section1, section2, section3, section4, section5].mutableCopy;
    }
    
    return self;
}

@end

File: Contents/SevenSwitch/SevenSwitch_iOS/Supporting Files/Model/ItemsForTableView.swift
//
//  ItemsForTableView.swift
//  SevenSwitch_iOS
//
//  Created by Kwan Hyun Son on 2022/10/10.
//

import Foundation

typealias SectionItems = (sectionTitle: String, items: Array<Item>)

final class ItemsForTableView {
    
    // MARK: - Property
    static let shared = ItemsForTableView()
    private(set) var allItems : Array<SectionItems>
    
    // MARK: - 생성 & 소멸
    private init() {
        
        let item1 = Item(title:"MGUSevenSwitch", detailText:"UISwitch와 같은 모양 연출가능하다.")
        let item2 = Item(title:"MGUSevenSwitch", detailText:"MGUSevenSwitch various examples")
        let item3 = Item(title:"Mini Timer 앱 사용예", detailText:"Mini Timer 앱에서 사용한 스위치를 살펴보자.")

        allItems = [(sectionTitle: "Basic", items: [item1]),
                    (sectionTitle: "advanced", items: [item2]),
                    (sectionTitle: "Using Sample", items: [item3])]
    }
}
